### ISSUE TITLE
[BUG] `.copy` method seems not to work properly

### ISSUE THREAD

**Issue by xxsl** [fireducks-dev/fireducks#10]
We all know that FireDucks does not support `.copy` with `deep=False` currently, this is okay.
But it seems that `deep=True` is not working either.

So what I mean is that the whole `.copy` method is ***completely broken***.

Let's see this case:

```python3
import fireducks.pandas as pd
sample_data = [["A", 1]]
df = pd.DataFrame(sample_data, columns = ["name", "value"])

# df_copy1 = df.copy(deep=False) # <--- not support currently so it's okay to skip it

# deep copy to df_copy2
df_copy2 = df.copy(deep=True)

# copy without parameter default to deep=True as Pandas api reference says (https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.DataFrame.copy.html)
df_copy3 = df.copy() # this is the same as deep=True, because parameter `deep` default is True

# modify it
df.iloc[0, 1] = 3

###
# so then we should get df value: A 3
print(df)
#  name  value
#     A      3   <--- This is expected

###
# df_copy2 and df_copy3 should be: A 1, as we use deep copy
print(df_copy2)
#  name  value
#     A      3   <--- This is wrong, should be 1

###
print(df_copy3)
#  name  value
#     A      3   <--- This is wrong, should be 1
```

![2024-03-04_15-34](https://github.com/fireducks-dev/fireducks/assets/7067381/2ef52d7d-6cd8-4a2b-a4b8-ad0afc9abca3)

We could see original Pandas handling this well:

![2024-03-04_15-37](https://github.com/fireducks-dev/fireducks/assets/7067381/ea591f98-3053-43c0-8302-25fef01a6099)

Tested on:
- FireDucks version: 0.10.2
- OS: Ubuntu 22.04.3 LTS (GNU/Linux 6.1.0-17-amd64 x86_64)
- Python3 version: Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux

**Comment by qsourav** [comment]
Thank you for reporting the issue.

The main problem was in lazy-execution with iloc-assignment. 

The copy(deep=True) will work for other case, for example instead of iloc-assignment, if you try: 
    df["value"] = df["value"] + 2
    
It would work as expected. 

Anyhow, we have fixed the lazy-execution issue with iloc-assignment and it will be released in our next version sometime in the next week.

**Comment by xxsl** [comment]
Hi @qsourav,

Thanks for your reply.

> The main problem was in lazy-execution with iloc-assignment.
> 
> The copy(deep=True) will work for other case, for example instead of iloc-assignment, if you try:
> 
> df["value"] = df["value"] + 2
> It would work as expected.


I can confirm what you said is correct.

![2024-03-07_17-23](https://github.com/fireducks-dev/fireducks/assets/7067381/3aeae462-cf94-4dfb-8046-3f2d37152078)

The problem seems to be only in iloc-assignment due to some issue of lazy-execution.

> Anyhow, we have fixed the lazy-execution issue with iloc-assignment and it will be released in our next version sometime in the next week.

Thanks very much for the help and your hard work. Look forward to the update.

Have a nice day.

**Comment by qsourav** [comment]
Hello @xxsl ,

We are pleased to inform you that the reported issue has been fixed in our latest released version 0.10.4.

We would appreciate if you kindly consider evaluating your case with FireDucks 0.10.4 and confirm the situation.

**Comment by xxsl** [comment]
Hi @qsourav 

I can confirm this issue has been fixed in 0.10.4.

![2024-03-14_17-14](https://github.com/fireducks-dev/fireducks/assets/7067381/0b1f37a4-ffb3-401a-aeb2-de71e2afeda0)

Thanks for your help.
